src/servers/src/mysql/handler.rs

```diff
@@ -134,18 +134,82 @@ impl MysqlInstanceShim {
         self.query_handler.do_describe(statement, query_ctx).await
     }
 
-    /// Save query and logical plan, return the unique id
-    fn save_plan(&self, plan: SqlPlan) -> u32 {
-        let stmt_id = self.prepared_stmts_counter.fetch_add(1, Ordering::Relaxed);
+    /// Save query and logical plan, return the unique key
+    fn save_plan(&self, plan: SqlPlan, stmt_key: String) {
         let mut prepared_stmts = self.prepared_stmts.write();
-        let _ = prepared_stmts.insert(stmt_id, plan);
-        stmt_id
+        let _ = prepared_stmts.insert(stmt_key, plan);
```

Comment:
Can you impl the [`DEALLOCATE PREPARE`](https://dev.mysql.com/doc/refman/8.4/en/deallocate-prepare.html), too? It's simple, just parse the stmt and remove the saved plan according to its carried stmt key. Otherwise the user's submitted prepared stmts have no way to be removed, always occupying memory.